gamemode/modules/base/sh_util.lua

```diff
@@ -56,29 +56,72 @@ end
 
 --[[---------------------------------------------------------------------------
 Find a player based on given information
+
+Note that there is a searching priority:
+  * UserID
+  * SteamID64
+  * SteamID
+  * Nick
+  * SteamName
+
+Note also that there are _separate_ loops. This is to make sure the function
+gives the same result, regardless of the order in which players are iterated
+over.
 ---------------------------------------------------------------------------]]
 function DarkRP.findPlayer(info)
     if not info or info == "" then return nil end
     local pls = player.GetAll()
 
-    for k = 1, #pls do -- Proven to be faster than pairs loop.
-        local v = pls[k]
-        if tonumber(info) == v:UserID() then
-            return v
+    local count = #pls
+    local numberInfo = tonumber(info)
+
+    -- First check if the input matches a player by UserID or SteamID64. This is
+    -- only necessary if the input can be parsed as a number.
+    if numberInfo then
+        for k = 1, count do
+            local v = pls[k]
+
+            if numberInfo == v:UserID() then
+                return v
+            end
         end
 
-        if info == v:SteamID() then
-            return v
+        for k = 1, count do
```

Comment:
A future optimization here could be to only run this loop when `info` is a valid SteamID64. SteamID64s have a specific format to them, and not nearly every int is a valid SteamID64.

A function `isValidSteamID64(some_number)` is a bit of a challenge to write, though, as it requires a thorough understanding of what makes a valid SteamID64.